DeveloperGuide: update implementation for 'PlannerList'

Letm3through opened the pull request:
Let's update DeveloperGuide to include 'PlannerList' which is to
be updated in the future.

chyeo commented on docs/DeveloperGuide.adoc:
```diff
@@ -310,6 +310,40 @@ image::UndoRedoActivityDiagram.png[width="650"]
 ** Cons: Requires dealing with commands that have already been undone: We must remember to skip these commands. Violates Single Responsibility Principle and Separation of Concerns as `HistoryManager` now needs to do two different things.
 // end::undoredo[]
 
+=== Planner List Feature
```
Could you add a tag in front of this section eg: `// tag::planner-list[]`

Letm3through replied:
Working on the part. Thanks for mentioning.

chyeo commented on docs/DeveloperGuide.adoc:
```diff
@@ -310,6 +310,40 @@ image::UndoRedoActivityDiagram.png[width="650"]
+* include `year` and `semester` attributes in one `planner_list` command and list degree planner(s)
+i.e. `planner_list y/YEAR s/SEMESTER`
+returns module having its year or semester matches the given year and semester
+
```
Could you add a tag at the end of this section as well?
eg: // end::planner-list[]

chyeo commented on docs/DeveloperGuide.adoc:
```diff
@@ -310,6 +310,40 @@ image::UndoRedoActivityDiagram.png[width="650"]
+.PlannerList component interactions
+image::PlannerListComponentSequenceDiagram.png[width="650"]
+
+==== Current Implementation
```
You might want to consider adding in some design considerations as well

truegitnovice replied:
Added one for now. Thank you

Letm3through committed:
DeveloperGuide: update implementation for 'PlannerList'

Let's update DeveloperGuide to include 'PlannerList' which is to
be updated in the future.

chyeo commented on docs/DeveloperGuide.adoc:
```diff
@@ -383,6 +383,49 @@ In addition to adding module code(s) to a requirement category, the `Requirement
+==== Design Considerations
+===== Aspect: How to parse multiple attributes
+
+- Alternative 1 (current choice): Parse the text related to each `PREFIX` individually
```
This `Alternative 1` is it compared to anything that lead you to chose this over the other?
I think that including the other alternatives would give the reader a better idea of why `Alternative 1` was chosen.

Letm3through replied:
Thanks for the pinpoint. Added additional alternative.

chyeo reviewed (approved):
LGTM. Thanks for updating the developer guide!
